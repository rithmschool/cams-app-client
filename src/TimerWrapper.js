import React, { Component } from "react";
import Timer from "./Timer";
import PropTypes from "prop-types";

class TimerWrapper extends Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: props.timer || 30,
      timerEnded: false
    };
    this.onEnd = this.onEnd.bind(this);
  }

  componentDidMount() {
    this.props.toggle();
  }

  onEnd() {
    this.props.toggle();
    this.setState({ timerEnded: true });
  }

  render() {
    let nextScreenMsg = this.state.timerEnded ? (
      <p className="lg">Press the space key to move on </p>
    ) : null;
    return (
      <div>
        <p>{this.props.title} </p>
        <p>Please describe what happened in the video</p>
        <Timer counter={this.state.counter} onEnd={this.onEnd} />
        {nextScreenMsg}
      </div>
    );
  }
}

TimerWrapper.propTypes = {
  toggle: PropTypes.func.isRequired,
  timer: PropTypes.number,
  title: PropTypes.string
};

export default TimerWrapper;
