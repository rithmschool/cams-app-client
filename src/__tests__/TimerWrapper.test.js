import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import TimerWrapper from "../TimerWrapper";
import toJson from "enzyme-to-json";

test("shapshot of TimerWrapper with timer prop", () => {
  const timer = renderer
    .create(<TimerWrapper timer={45} toggle={function() {}} />)
    .toJSON();
  expect(timer).toMatchSnapshot();
});

test("shapshot of TimerWrapper without timer prop", () => {
  const timer = renderer
    .create(<TimerWrapper toggle={function() {}} />)
    .toJSON();
  expect(timer).toMatchSnapshot();
});
