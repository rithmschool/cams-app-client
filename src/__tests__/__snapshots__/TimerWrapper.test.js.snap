// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shapshot of TimerWrapper with timer prop 1`] = `
<div>
  <div
    className="timer"
  >
    <h2
      className="center"
    >
       
      45
       
    </h2>
  </div>
</div>
`;

exports[`shapshot of TimerWrapper without timer prop 1`] = `
<div>
  <div
    className="timer"
  >
    <h2
      className="center"
    >
       
      30
       
    </h2>
  </div>
</div>
`;

exports[`snapshot of TimerWrapper when counter is set to zero 1`] = `
<div>
  <div
    className="timer"
  >
    <h2
      className="center"
    >
       
      0
       
    </h2>
  </div>
</div>
`;
